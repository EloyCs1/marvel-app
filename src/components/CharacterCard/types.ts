import { AppCharacter } from "../../service/types";

export interface CharacterCardProps {
  character: AppCharacter.Character;
}
