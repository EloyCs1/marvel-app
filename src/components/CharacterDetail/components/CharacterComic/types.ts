import { AppCharacter } from "src/service/types";

export interface CharacterComicProps {
  comic: AppCharacter.Comic;
}
