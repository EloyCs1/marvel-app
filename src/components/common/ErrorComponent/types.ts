export interface ErrorComponentProps {
  testId?: string;
  tryAgain?: () => void;
}
