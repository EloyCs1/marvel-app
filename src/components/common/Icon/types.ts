export enum IconName {
  FAVORITES_FILLED,
  FAVORITES_OUTLINED,
  SEARCH,
  SPINNER,
  MARVEL_LOGO,
}

export interface IconProps {
  testId?: string;
  name: IconName;
  width?: number;
  height?: number;
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}
