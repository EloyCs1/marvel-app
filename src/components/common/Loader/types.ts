export interface LoaderProps {
  testId?: string;
}
