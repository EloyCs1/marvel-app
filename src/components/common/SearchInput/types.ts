import React from "react";

export interface SearchInputProps {
  testId?: string;
  loading?: boolean;
  placeholder?: string;
  helpText?: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
