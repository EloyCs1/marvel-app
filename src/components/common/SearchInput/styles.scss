.searchInput {
  height: 77px;
  padding: 24px 0;
  gap: 12px;

  display: flex;
  flex-direction: column;

  &__input {
    display: flex;
    width: 100%;
    position: relative;

    svg {
      margin-top: 7.5px;
      position: absolute;
    }

    input {
      width: 100%;
      padding: 0px 0px 8px 24px;
      border: none;
      border-bottom: 1px solid var(--color-border);
      outline: none;
      font-size: 16px;
      height: 27px;
      text-transform: uppercase;

      &:disabled {
        background-color: transparent;
        border-bottom: 1px solid var(--color-disabled);
      }
    }
  }

  &__helpText {
    font-size: 12px;
    color: var(--color-black);
    text-transform: uppercase;
  }
}
