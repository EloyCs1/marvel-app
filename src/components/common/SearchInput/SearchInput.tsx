import Icon from "../Icon/Icon";
import { IconName } from "../Icon/types";

import { SearchInputProps } from "./types";
import "./styles.scss";

const SearchInput: React.FC<SearchInputProps> = ({
  testId = "searchInput",
  placeholder,
  loading = false,
  helpText,
  value,
  onChange,
}) => {
  return (
    <div className="searchInput" data-testid={testId}>
      <div className="searchInput__input">
        <Icon testId={`${testId}-icon`} width={12} height={12} name={loading ? IconName.SPINNER : IconName.SEARCH} />
        <input
          type="text"
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          disabled={loading}
          data-testid={`${testId}-input`}
        />
      </div>
      {helpText && (
        <p className="searchInput__helpText" data-testid={`${testId}-helpText`}>
          {helpText}
        </p>
      )}
    </div>
  );
};
export default SearchInput;
