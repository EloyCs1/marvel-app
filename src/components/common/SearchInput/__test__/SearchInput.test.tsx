import { render } from "@testing-library/react";

import { AppWrapper } from "src/__test__/test.utils";

import { SearchInputProps } from "../types";
import SearchInput from "../SearchInput";

const defaultProps: SearchInputProps = {
  value: "",
  onChange: jest.fn(),
};

describe("SearchInput", () => {
  test("Default render", () => {
    const wrapper = render(
      <AppWrapper>
        <SearchInput {...defaultProps} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("Loading render", () => {
    const wrapper = render(
      <AppWrapper>
        <SearchInput {...defaultProps} loading={true} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("helpText render", () => {
    const wrapper = render(
      <AppWrapper>
        <SearchInput {...defaultProps} helpText={"helpText"} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("value and placeholder render", () => {
    const wrapper = render(
      <AppWrapper>
        <SearchInput {...defaultProps} value={"value"} placeholder={"placeholder"} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
