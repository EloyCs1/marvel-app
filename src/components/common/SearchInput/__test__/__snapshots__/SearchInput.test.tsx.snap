// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Default render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="searchInput"
        data-testid="searchInput"
      >
        <div
          class="searchInput__input"
        >
          <svg
            data-testid="searchInput-icon"
            fill="none"
            height="12"
            viewBox="0 0 13 13"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
              fill="black"
            />
          </svg>
          <input
            data-testid="searchInput-input"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="searchInput"
      data-testid="searchInput"
    >
      <div
        class="searchInput__input"
      >
        <svg
          data-testid="searchInput-icon"
          fill="none"
          height="12"
          viewBox="0 0 13 13"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
            fill="black"
          />
        </svg>
        <input
          data-testid="searchInput-input"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchInput Loading render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="searchInput"
        data-testid="searchInput"
      >
        <div
          class="searchInput__input"
        >
          <svg
            data-testid="searchInput-icon"
            height="12"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="iconSpinner"
              d="M10.14,1.16a11,11,0,0,0-9,8.92A1.59,1.59,0,0,0,2.46,12,1.52,1.52,0,0,0,4.11,10.7a8,8,0,0,1,6.66-6.61A1.42,1.42,0,0,0,12,2.69h0A1.57,1.57,0,0,0,10.14,1.16Z"
              fill="black"
            />
          </svg>
          <input
            data-testid="searchInput-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="searchInput"
      data-testid="searchInput"
    >
      <div
        class="searchInput__input"
      >
        <svg
          data-testid="searchInput-icon"
          height="12"
          viewBox="0 0 24 24"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="iconSpinner"
            d="M10.14,1.16a11,11,0,0,0-9,8.92A1.59,1.59,0,0,0,2.46,12,1.52,1.52,0,0,0,4.11,10.7a8,8,0,0,1,6.66-6.61A1.42,1.42,0,0,0,12,2.69h0A1.57,1.57,0,0,0,10.14,1.16Z"
            fill="black"
          />
        </svg>
        <input
          data-testid="searchInput-input"
          disabled=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchInput helpText render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="searchInput"
        data-testid="searchInput"
      >
        <div
          class="searchInput__input"
        >
          <svg
            data-testid="searchInput-icon"
            fill="none"
            height="12"
            viewBox="0 0 13 13"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
              fill="black"
            />
          </svg>
          <input
            data-testid="searchInput-input"
            type="text"
            value=""
          />
        </div>
        <p
          class="searchInput__helpText"
          data-testid="searchInput-helpText"
        >
          helpText
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="searchInput"
      data-testid="searchInput"
    >
      <div
        class="searchInput__input"
      >
        <svg
          data-testid="searchInput-icon"
          fill="none"
          height="12"
          viewBox="0 0 13 13"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
            fill="black"
          />
        </svg>
        <input
          data-testid="searchInput-input"
          type="text"
          value=""
        />
      </div>
      <p
        class="searchInput__helpText"
        data-testid="searchInput-helpText"
      >
        helpText
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchInput value and placeholder render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="searchInput"
        data-testid="searchInput"
      >
        <div
          class="searchInput__input"
        >
          <svg
            data-testid="searchInput-icon"
            fill="none"
            height="12"
            viewBox="0 0 13 13"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
              fill="black"
            />
          </svg>
          <input
            data-testid="searchInput-input"
            placeholder="placeholder"
            type="text"
            value="value"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="searchInput"
      data-testid="searchInput"
    >
      <div
        class="searchInput__input"
      >
        <svg
          data-testid="searchInput-icon"
          fill="none"
          height="12"
          viewBox="0 0 13 13"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.9062 10.9922C12.0234 11.1094 12.0234 11.2969 11.9062 11.3906L11.3672 11.9297C11.2734 12.0469 11.0859 12.0469 10.9688 11.9297L8.13281 9.09375C8.08594 9.02344 8.0625 8.95312 8.0625 8.88281V8.57812C7.19531 9.30469 6.09375 9.75 4.875 9.75C2.17969 9.75 0 7.57031 0 4.875C0 2.20312 2.17969 0 4.875 0C7.54688 0 9.75 2.20312 9.75 4.875C9.75 6.09375 9.28125 7.21875 8.55469 8.0625H8.85938C8.92969 8.0625 9 8.10938 9.07031 8.15625L11.9062 10.9922ZM4.875 8.625C6.9375 8.625 8.625 6.96094 8.625 4.875C8.625 2.8125 6.9375 1.125 4.875 1.125C2.78906 1.125 1.125 2.8125 1.125 4.875C1.125 6.96094 2.78906 8.625 4.875 8.625Z"
            fill="black"
          />
        </svg>
        <input
          data-testid="searchInput-input"
          placeholder="placeholder"
          type="text"
          value="value"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
